lib/features/email/presentation/email_controller.dart

```diff
@@ -67,13 +80,41 @@ class EmailController extends BaseController {
     emailAddressExpandMode.value = expandMode;
   }
 
+  void markAsEmailRead(BuildContext context, PresentationEmail presentationEmail, ReadActions readActions) async {
+    final accountId = mailboxDashBoardController.accountId.value;
+    final mailboxCurrent = mailboxDashBoardController.selectedMailbox.value;
+    if (accountId != null && mailboxCurrent != null) {
+      _markAsEmailReadInteractor.execute(accountId, presentationEmail.id, readActions)
+          .then((result) => result.fold(
+              (failure) {
+                if (readActions == ReadActions.markAsUnread) {
+                  backToThreadView(context);
+                }},
+              (success) {
+                if (success is MarkAsEmailReadSuccess) {
+                  emailActionCurrent = MarkAsEmailReadAndUnreadAction();
+                  _updateMailboxDashBoardActionForDesktop(context, emailActionCurrent);
+                }
+                if (readActions == ReadActions.markAsUnread) {
+                  backToThreadView(context);
+                }
+              }));
```

Comment:
quite hard to understand, it is break our agreement on unidirectional data flow. `viewState` is not update correctly base on the change of the new state. Please take care the `viewState`.

Reply:
What you say is quite confusing. Please describe in detail what you want.

Reply:
please read it here: https://github.com/linagora/tmail-flutter/pull/93#discussion_r709725474